arch/x86/soc/intel_quark/quark_se/soc.h

```diff
@@ -89,6 +90,10 @@
 
 #endif /*  _ASMLANGUAGE */
 
+void usb_dw_clock_disable(void);
+int usb_dw_clock_enable(void);
+void _usb_dw_int_unmask(void);
```

Comment:
The driver will fail if someone tries to build it for anything but quark se and there is no reference or documentation of how those functions you are moving are defined.

Reply:
I will add a README  in "drivers/usb/devices/ " explaining that any Soc which wants to use USB designware driver  would have to define the above apis in their soc.c and prototype in soc.h. As of now only quark_se is only Soc which using designware USB IP.